src/components/Swap/SwapActionButton/SwapButton.tsx

```diff
@@ -29,28 +33,37 @@ export default function SwapButton({
 }: {
   color: keyof Colors
   disabled: boolean
-  onSubmit: (submit: () => Promise<ApprovalTransactionInfo | SwapTransactionInfo | undefined>) => Promise<boolean>
+  onSubmit: (submit?: () => Promise<ApprovalTransactionInfo | SwapTransactionInfo | void>) => Promise<void>
 }) {
   const { account, chainId } = useWeb3React()
   const {
     [Field.INPUT]: { usdc: inputUSDC },
     [Field.OUTPUT]: { usdc: outputUSDC },
     trade: { trade, gasUseEstimateUSD },
+    approval,
+    permit,
     slippage,
     impact,
-    approval,
   } = useSwapInfo()
   const deadline = useTransactionDeadline()
   const feeOptions = useAtomValue(feeOptionsAtom)
 
-  const { callback: swapCallback } = useSwapCallback({
-    trade,
+  const permit2 = usePermit2()
+  const { callback: swapRouterCallback } = useSwapCallback({
```

Comment:
we may wanna rename this hook now to `useAutoRouterCallback`?

Reply:
I don't want to touch any of the old code. I'm hoping to just gut it in the very near future.